src/Plugin/StrawberryRunnersPostProcessor/OcrPostProcessor.php

```diff
@@ -0,0 +1,350 @@
+    $arguments_gs = $config['arguments'];
+    $execpath_tesseract = $config['path_tesseract'];
+    $arguments_tesseract = $config['arguments_tesseract'];
+
+    if (empty($file_path)) {
+      return NULL;
+    }
+
+    // This run function executes a 2 step function
+    //-- with r300 == 300dpi, should be configurable, etc. All should be configurable
+    // First gs -dBATCH -dNOPAUSE -sDEVICE=pnggray -r300 -dUseCropBox -sOutputFile=somepage_pagenumber.png %file
+
+    $command = '';
+    $can_run_gs = \Drupal::service('strawberryfield.utility')->verifyCommand($execpath_gs);
+    $can_run_tesseract = \Drupal::service('strawberryfield.utility')->verifyCommand($execpath_tesseract);
+    $filename = pathinfo($file_path, PATHINFO_FILENAME);
+    $sourcefolder =  pathinfo($file_path,PATHINFO_DIRNAME);
+    $sourcefolder = strlen($sourcefolder)> 0 ? $sourcefolder.'/' : sys_get_temp_dir().'/';
+    $gs_destination_filename = "{$sourcefolder}{$filename}_{$page_number}.png";
+    if ($can_run_gs &&
+      $can_run_tesseract &&
+      (strpos($arguments_gs, '%file' ) !== FALSE) &&
+      (strpos($arguments_tesseract, '%file' ) !== FALSE)) {
+      $arguments_gs = "-dBATCH -dNOPAUSE -r300 -dUseCropBox -dQUIET -sDEVICE=pnggray -dFirstPage={$page_number} -dLastPage={$page_number} -sOutputFile=$gs_destination_filename " . $arguments_gs;
+      $arguments_gs = str_replace('%s','', $arguments_gs);
+      $arguments_gs = str_replace_first('%file','%s', $arguments_gs);
+      $arguments_gs = sprintf($arguments_gs, $file_path);
+
+      $arguments_tesseract = str_replace('%s','', $arguments_tesseract);
+      $arguments_tesseract = str_replace_first('%file','%s', $arguments_tesseract);
+      $arguments_tesseract = sprintf($arguments_tesseract, $gs_destination_filename);
+
+      $command_gs = escapeshellcmd($execpath_gs.' '.$arguments_gs);
+      $command_tesseract = escapeshellcmd($execpath_tesseract.' '.$arguments_tesseract);
+
+      $command = $command_gs.' && '.$command_tesseract;
+
+    } else {
+      error_log("missing arguments for OCR");
+    }
+    // Only return $command if it contains the original filepath somewhere
+    if (strpos($command, $file_path) !== false) { return $command;}
+    return '';
+
+  }
+
+  protected function hOCRtoMiniOCR($output, $pageid) {
+    error_log($output);
+    $hocr = simplexml_load_string($output);
+    $internalErrors = libxml_use_internal_errors(TRUE);
+    libxml_clear_errors();
+    libxml_use_internal_errors($internalErrors);
+    if (!$hocr) {
+      error_log('Could not convert HOCR to MiniOCR, sources is not valid XML');
+      return NULL;
+    }
+    $w = new \XMLWriter();
+    $w->openMemory();
+    $w->startDocument('1.0','UTF-8');
+    $w->startElement("ocr");
+    foreach ($hocr->body->children() as $page) {
+      $coos = explode(" ", substr($page['title'], 5));
+      if (count($coos)) {
+        $w->startElement("p");
+        $w->writeAttribute("id", $pageid);
```

Comment:
@DiegoPino If I remember well, here we need xml:id as state in a not documented issue in plugin

Reply:
OK. Is that a bug of the plugin? Still on the xmlwriter code I wrote I don't seem to be able to output the 'text' of the elements out.  This whole part needs 1-2 hours more of code. Will see if I can add a namespace id and will let you know.

Reply:
It is not a bug, I understood only not well documented. Now appears:
![image](https://user-images.githubusercontent.com/4232178/99974426-352c1c80-2da1-11eb-834e-58e9531bae71.png)